onnxruntime/core/providers/cuda/tensor/gather_elements_impl.cu

```diff
@@ -1,140 +1,291 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include "core/providers/cuda/tensor/gather_elements_impl.h"
+#include "core/providers/cuda/tensor/scatter_elements_impl.h"
+#ifdef ENABLE_TRAINING
+#include "orttraining/training_ops/cuda/tensor/gather_elements_grad_impl.h"
+#endif
+
+#include "core/providers/cuda/atomic/common.cuh"
 #include "core/providers/cuda/cu_inc/common.cuh"
-#include "gather_elements_impl.h"
 
 namespace onnxruntime {
 namespace cuda {
 
 namespace {
-constexpr int threads_per_block = GridDim::maxThreadsPerBlock;
-constexpr int thread_worksize = 16;
+constexpr int kThreadsPerBlock = GridDim::maxThreadsPerBlock;
+constexpr int kThreadWorkSize = 4;
```

Comment:
Does kThreadWorkSize change bring better perf or both 2D and other cases, on both V100 and A100? Do you know what the possible reason a light-weighter kernel bring a better perf?